cli/cmds/cluster_create.go

```diff
@@ -245,3 +262,47 @@ func waitForCluster(ctx context.Context, client client.Client, cluster *v1alpha1
 		return false, nil
 	})
 }
+
+func CreateCustomCertsSecret(ctx context.Context, name, namespace, customCertsPath string, client client.Client) error {
+	var customCertSecret = v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      controller.SafeConcatNameWithPrefix(name, "custom", "certs"),
+			Namespace: namespace,
+		},
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Secret",
+			APIVersion: "v1",
+		},
+		Data: map[string][]byte{},
+	}
+
+	err := filepath.Walk(customCertsPath, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if info.IsDir() {
+			return nil
+		}
+
+		prefix := ""
+		if strings.Contains(path, "etcd") {
+			prefix = "etcd-"
+		}
+
+		fileContent, err := os.ReadFile(path)
+		if err != nil {
+			return err
+		}
+
+		customCertSecret.Data[prefix+info.Name()] = fileContent
+
+		return nil
+	})
```

Comment:
The `Walk` func can be useful when you need to recursively iterate into the subdirectories, otherwise a ReadDir is probably simpler:

```go
files, err := os.ReadDir(customCertsPath)
if err != nil {
	return err
}

for _, file := range files {
	if file.IsDir() {
		continue
	}

	filePath := filepath.Join(customCertsPath, file.Name())
	fileContent, err := os.ReadFile(filePath)
	if err != nil {
		return err
	}
	
	// etc
}
```